financial-connections/src/main/java/com/stripe/android/financialconnections/repository/ConsumerSessionRepository.kt

```diff
@@ -36,15 +39,20 @@ internal class RealConsumerSessionRepository @Inject constructor(
 
     override fun storeConsumerSession(
         consumerSession: ConsumerSession?,
+        publishableKey: String?,
     ) {
-        savedStateHandle[KeyConsumerSession] = consumerSession?.toCached()
+        val existingSession = provideConsumerSession()
+        val newPublishableKey = publishableKey ?: existingSession?.publishableKey
+        savedStateHandle[KeyConsumerSession] = consumerSession?.toCached(newPublishableKey)
```

Comment:
[suggestion] to avoid this, we could have a wrapper model to keep the same hierarchy as the remote model. 

`ConsumerSessionState(val publishableKey: String, val session: CachedConsumerSession)`